hack/boilerplate/boilerplate.py

```diff
@@ -183,19 +204,25 @@ def get_files(extensions):
     return outfiles
 
 
+def get_dates():
+    years = datetime.datetime.now().year
+    return '(%s)' % '|'.join((str(year) for year in range(2014, years+1)))
+
+
 def get_regexs():
     regexs = {}
     # Search for "YEAR" which exists in the boilerplate, but shouldn't in the real thing
     regexs["year"] = re.compile('YEAR')
-    # dates can be 2014, 2015, 2016, ..., CURRENT_YEAR, company holder names can be anything
-    years = range(2014, date.today().year + 1)
-    regexs["date"] = re.compile(
-        '(%s)' % "|".join(map(lambda l: str(l), years)))
+    # get_dates return 2014, 2015, 2016, 2017, or 2018 until the current year as a regex like: "(2014|2015|2016|2017|2018)";
+    # company holder names can be anything
+    regexs["date"] = re.compile(get_dates())
     # strip // +build \n\n build constraints
     regexs["go_build_constraints"] = re.compile(
         r"^(// \+build.*\n)+\n", re.MULTILINE)
-    # strip #!.* from shell scripts
+    # strip #!.* from scripts
     regexs["shebang"] = re.compile(r"^(#!.*\n)\n*", re.MULTILINE)
+    # Search for generated files
+    regexs["generated"] = re.compile('DO NOT EDIT')
     return regexs
 
 
```

Comment:
Rob, a lot of changes in this file seem unrelated. Did you copy this _almost_ verbatim from somewhere specific in upstream? If yes, should we include a comment at the top of the file to point to the original source for future-selves?

Reply:
Good call, I've added a comment at the top of this file with a link to the specific upstream file this was copied from: https://github.com/kubernetes/kubernetes/blob/04c2b1fbdc1289c9a72eda87cf7072346e60d241/hack/boilerplate/boilerplate.py